src/components/ContactsTable.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import shortid from 'shortid';
+import TableRow from './TableRow';
+
+
+class ContactsTable extends React.Component {
+  //  we use `constructor` as we are setting initial `state` value from the `props`
+  constructor(props) {
+    super();
+    this.state = {
+      contacts: props.contactsArray.slice(0, 5)
+    }
+  }
+
+  addContact = () => {
+    const randomNumber = Math.floor((Math.random() * this.props.contactsArray.length - 1) + 5);
+    this.state.contacts.push(this.props.contactsArray[randomNumber])
+    this.setState({ contacts: this.state.contacts });
+  }
+
+
+  deleteContact = (contactIndex) => {
+    //console.log("DELETE", contactIndex)
+
+    this.state.contacts.splice(contactIndex, 1);
+    this.setState({ contacts: this.state.contacts });
+  }
+
+  sortContactsByName = () => {
+
+
+    // sort by name
+    const sortedArray = this.state.contacts.sort(function (a, b) {
+      let nameA = a.name.toUpperCase(); // ignore upper and lowercase
+      let nameB = b.name.toUpperCase(); // ignore upper and lowercase
+      if (nameA < nameB) {
+        return -1;
+      }
+      if (nameA > nameB) {
+        return 1;
+      }
+      return 0;
+    });
+
+    this.setState({ contacts: sortedArray });
+  }
+
+
+  sortContactsByPopularity = () => {
+
+    // sort by popularity
+    const sortedArray = this.state.contacts.sort(function (a, b) {
+      return a.popularity - b.popularity;
```

Comment:
instead of having to use `reverse()` after you could do `b.popularity - a.popularity`